### PR TITLE
copr: Add vectorize trim1Arg

### PR DESCRIPTION
**Pull request by Poytr1**
UCP #6957
Signed-off-by: Poytr1 <pooytr1@gmail.com>

###  What have you changed?

Add vectorize trim1Arg

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No
###  Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/issues/6957

### CODE REVIEW

**Comment by Poytr1** [comment]
Hi @sticnarf , could you please help me review the pr? Thanks.

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 1]
```diff
@@ -409,6 +409,18 @@ pub fn find_in_set(s: &Option<Bytes>, str_list: &Option<Bytes>) -> Result<Option
     })
 }
 
+#[rpn_fn]
+#[inline]
+pub fn trim_1_arg(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    Ok(arg.as_ref().map(|bytes| {
+        let s = String::from_utf8_lossy(bytes);
+        s.trim_start_matches(' ')
+            .trim_end_matches(' ')
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: I think you can simply use `trim_matches`

**Comment by sticnarf** [comment]
The PR title needs correction.You are implementing the vectorized version :)

File: components/tidb_query/src/rpn_expr/impl_string.rs (hunk 1 above)
Comment (Poytr1) [reply components/tidb_query/src/rpn_expr/impl_string.rs]: Good catch, done.

**Comment by Poytr1** [comment]
> The PR title needs correction.You are implementing the vectorized version :)

Yeah..you are right, just corrected the title. Thanks😆

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 2]
```diff
@@ -409,6 +409,15 @@ pub fn find_in_set(s: &Option<Bytes>, str_list: &Option<Bytes>) -> Result<Option
     })
 }
 
+#[rpn_fn]
+#[inline]
+pub fn trim_1_arg(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    Ok(arg.as_ref().map(|bytes| {
+        let s = String::from_utf8_lossy(bytes);
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: Here we don't need to allocate a new string. Using `std::str::from_utf8` can save an allocation.

Comment (Poytr1) [reply components/tidb_query/src/rpn_expr/impl_string.rs]: Done.

**Comment by sticnarf** [comment]
@Poytr1 Really sorry. My fault.
I think `trim` needs to support invalid UTF-8 bytes. You should take `ltrim` or `rtrim` as a reference.

**Comment by Poytr1** [comment]
@sticnarf No worries, I just re-implemented the `trim_1_arg` function, please take a look.

**Comment by Poytr1** [comment]
Hi @sticnarf , could you please help me review this change again?

**Comment by andylokandy** [comment]
@Poytr1 FYI:
https://github.com/tikv/tikv/blob/f6772aeaf6994ad37160f3b52d3e88e9e50ffb9b/components/tidb_query/src/rpn_expr/impl_string.rs#L934

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 3]
```diff
@@ -1625,6 +1639,37 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_trim_1_arg() {
+        let test_cases = vec![
+            (None, None),
```
[the remark this answers is not in the document]
Comment (Poytr1) [reply components/tidb_query/src/rpn_expr/impl_string.rs]: Done. PTAL

**Pull request merged**